File: Data Structures & Algorithms/majority-element-ii/submission-11.java
class Solution {
    public List<Integer> majorityElement(int[] nums) {
        int n = nums.length;
        int t = n / 3;
        Map<Integer, Integer> map = new HashMap<>();
        List<Integer> res = new ArrayList<>();
        for(int num : nums) {
            // map.put(num, map.getOrDefault(num, 0) + 1);
            if(!map.containsKey(num)) {
               map.put(num, 1);
            }
            else {
               map.put(num, map.get(num) + 1);
            }
        }
        for(Map.Entry<Integer, Integer> it : map.entrySet()) {
            if(it.getValue() > t) {
               res.add(it.getKey());
            }
        }
        return res;
    }
}

File: Data Structures & Algorithms/first-missing-positive/submission-11.cpp
class Solution {
public:
    int firstMissingPositive(vector<int>& nums) {
        sort(nums.begin(), nums.end());
        int res = 1;
        for(int num : nums) {
            if(num > 0 && res == num) {
                res++;
            }
        }
        return res;
    }
};